packages/platform-sdk/src/coins/network.models.ts

```diff
@@ -17,158 +17,153 @@ export type CoinTransactionTypes =
 
 export type ExpirationType = "height" | "timestamp";
 
-export interface SignatureMethods {
-	default: boolean;
-	ledgerS?: boolean;
-	ledgerX?: boolean;
+export type SignatureMethod = "default" | "musig" | "ledgerS" | "ledgerX";
+
+export interface NetworkHost {
+	type: "full" | "musig" | "archival" | "explorer";
+	host: string;
 }
 
-export interface NetworkFeatureFlags {
-	Client?: {
-		transaction?: boolean;
-		transactions?: boolean;
-		wallet?: boolean;
-		wallets?: boolean;
-		delegate?: boolean;
-		delegates?: boolean;
-		votes?: boolean;
-		voters?: boolean;
-		configuration?: boolean;
-		fees?: boolean;
-		syncing?: boolean;
-		broadcast?: boolean;
-	};
-	Fee?: {
-		all?: boolean;
-	};
-	Identity?: {
-		address?: {
-			mnemonic?: boolean;
-			multiSignature?: boolean;
-			publicKey?: boolean;
-			privateKey?: boolean;
-			wif?: boolean;
-			secret?: boolean;
-			validate?: boolean;
-		};
-		publicKey?: {
-			mnemonic?: boolean;
-			multiSignature?: boolean;
-			wif?: boolean;
-			secret?: boolean;
-		};
-		privateKey?: {
-			mnemonic?: boolean;
-			wif?: boolean;
-			secret?: boolean;
-		};
-		wif?: {
-			mnemonic?: boolean;
-			secret?: boolean;
-		};
-		keyPair?: {
-			mnemonic?: boolean;
-			privateKey?: boolean;
-			wif?: boolean;
-			secret?: boolean;
-		};
-	};
-	Ledger?: {
-		getVersion?: boolean;
-		getPublicKey?: boolean;
-		signTransaction?: boolean;
-		signMessage?: boolean;
-	};
-	Link?: {
-		block?: boolean;
-		transaction?: boolean;
-		wallet?: boolean;
-	};
-	Message?: {
-		sign?: boolean;
-		verify?: boolean;
-	};
-	Peer?: {
-		search?: boolean;
-	};
-	Transaction?: {
-		transfer?: SignatureMethods;
-		secondSignature?: SignatureMethods;
-		delegateRegistration?: SignatureMethods;
-		vote?: SignatureMethods;
-		multiSignature?: SignatureMethods;
-		ipfs?: SignatureMethods;
-		multiPayment?: SignatureMethods;
-		delegateResignation?: SignatureMethods;
-		htlcLock?: SignatureMethods;
-		htlcClaim?: SignatureMethods;
-		htlcRefund?: SignatureMethods;
-	};
-	Miscellaneous?: {
-		dynamicFees?: boolean;
-		memo?: boolean;
-		utxo?: boolean;
-	};
-	Derivation?: {
-		bip39?: boolean;
-		bip44?: boolean;
-		bip49?: boolean;
-		bip84?: boolean;
-		secret?: boolean;
-	};
-	Internal?: {
-		fastDelegateSync?: boolean;
-	};
+export type WalletPermission = "read" | "write";
+
+export interface ImportMethod {
+	default: boolean;
+	permissions: WalletPermission[];
 }
 
+export type MnemonicMethod = "bip39" | "bip44" | "bip49" | "bip84";
+
 export interface NetworkManifest {
```

Comment:
Without context, NetworkManifest can be initially understood as "internet network" (which I actually thought the first time), but I assume it means more like "blockchain network"? I was ok with CoinManifest. Or else BlockchainManifest?

Reply:
Those manifests are for blockchain networks, so `NetworkManifest` should make that obvious I would say. I'm not sure how I would end up thinking of a network connection in the context of the whole project 🤔

Reply:
Almost every package does some form of http/rpc so it's natural to think about network connections. Especially that most people use IDEs and autocompletes of general terms can bring you to wrong places. But anyway it's no big deal

Reply:
Yeah but if I go by that assumption I would have to rename a lot of things just under the assumption that someone could think about an HTTP connection and I generally avoid binding my naming conventions and development to the assumption that someone will open a project in a software that will make assumptions based on names that are typed out.

Reply:
Ok then it makes sense.
I still think that the word "Blockchain" should have a more central presence in the type system